### PR TITLE
fix: allow manual address for unsupported buy asset

### PR DESCRIPTION
**Pull request by 0xApotheosis**
## Description

Fixes an issue where we were early exiting and not invalidating tags in the `useGetTradeQuotes` hook when no receive address was available (i.e. the connected wallet did not support the buy asset).

This prevented the manual receive address input field from showing (unless switching from a supported sell asset, in which case the `skipToken` would incorrectly _not_ be set as we still have a `receiveAddress` at the time that code is run (before we invalidate the tags).

Examples where this would occur include trading into Avalanche on KeepKey, or into a UTXO on MetaMask (without snaps).

## Pull Request Type

- [x] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [ ] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

Closes https://github.com/shapeshift/web/issues/6393

High. Despite the small surface area this PR touches trade quote args in what should be a predictable way, but could have unintended side effects if something else is not behaving as expected that relied on the tags not being invalidated.

> What protocols, transaction types or contract interactions might be affected by this PR?

Affects quotes for all trades, which thus affects the trades themselves.

## Testing

- When connected to a KeepKey and trading into an asset on Avalanche we should have the ability to manually enter an Avalanche address.
- When connected to MetaMask (without snaps) and trading into an unsupported asset (e.g. a UTXO) we should have the ability to manually enter an appropriate UTXO address.
- All other trades should remain unaffected, as behalf as they do in production.

### Engineering

☝️

### Operations

☝️

## Screenshots (if applicable)

**Develop**

Connected to MetaMask without snaps we cannot trade into a UTXO as we can't enter an address manually:

Screenshot 2024-03-14 at 12 11 24 PM

### CODE REVIEW

File: src/components/MultiHopTrade/hooks/useGetTradeQuotes/useGetTradeQuotes.tsx [hunk 1]
```diff
@@ -162,6 +162,16 @@ export const useGetTradeQuotes = () => {
...
   useEffect(() => {
+    // Always invalidate tags when this effect runs - args have changed, and whether we want to fetch an actual quote
+    // or a "skipToken" no-op, we always want to ensure that the tags are invalidated before a new query is ran
+    // That effectively means we'll unsubscribe to queries, considering them stale
+    dispatch(swapperApi.util.invalidateTags(['TradeQuote']))
+
+    // Clear the slice before asynchronously generating the input and running the request.
+    // This is to ensure the initial state change is done synchronously to prevent race conditions
+    // and losing sync on loading state etc.
+    dispatch(tradeQuoteSlice.actions.clear())
```
Comment (0xApotheosis) [inline src/components/MultiHopTrade/hooks/useGetTradeQuotes/useGetTradeQuotes.tsx:173]: This eventually ran either way, so DRY it.

File: src/components/MultiHopTrade/components/TradeInput/TradeInput.tsx [hunk 2]
```diff
@@ -292,7 +294,7 @@ export const TradeInput = memo(({ isCompact }: TradeInputProps) => {
 
   const isLoading = useMemo(
     () =>
-      !isAnyTradeQuoteLoaded ||
+      (!isAnyTradeQuoteLoaded && !isTradeQuoteRequestAborted) ||
```
Comment (0xApotheosis) [inline src/components/MultiHopTrade/components/TradeInput/TradeInput.tsx:297]: If we have no quotes, but the trade quote request is aborted, we aren't actually still loading.

**Review by gomesalexandre — APPROVED** [review]
@0xApotheosis I wasn't able to repro the wrong behavior in develop, though I noticed this tended to be quite janky i.e first load vs. subsequent swapper loads which could explain why.

Tested against develop (right):

### Unsupported buy asset manual receive address is still showing up on initial selection of an unsupported asset

- Manual receive address is happy

image

Note, we're not considering things as loading anymore... because they're indeed not.

- Quote(s) are fetched on manual receive address input

image

- Clearing the manual receive address puts the user back in the "Input manual receive address" flow with no quotes being shown, and no loading state wrongly shown

image

### Unsupported buy asset manual receive address is not showing up when going from unsupported -> supported

image

### Unsupported buy asset manual receive address is showing up when going from unsupported -> supported

image

### Debouncing is still happy

i.e from 0.01 to 0.0123456

image

### Navigating to an unsupported sell asset route triggers the manual receive address input in swapper

image

### Navigating to an unsupported sell asset route *doesn't* trigger the manual receive address input in swapper

image

### Trading into KK on AVAX triggers the manual receive address input

image

**Review by woodenfurniture — APPROVED** [review]
Tested on metamask without snap installed. Trades with and without manual receive address are worky:

image

image

Confirmed funds were received at the destination address (took an app refresh after switching wallet:
image

**Pull request merged**